pkg/controller/handler.go

```diff
@@ -1,75 +1,46 @@
 package controller
 
 import (
-	"github.com/containous/maesh/pkg/k8s"
 	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
 )
 
+// configMessageChanRebuild rebuild.
+var configMessageChanRebuild = struct{}{}
+
 // Handler is an implementation of a ResourceEventHandler.
 type Handler struct {
 	log               logrus.FieldLogger
-	ignored           k8s.IgnoreWrapper
-	configRefreshChan chan string
+	configRefreshChan chan struct{}
 	serviceManager    ServiceManager
 }
 
 // NewHandler creates a handler.
-func NewHandler(log logrus.FieldLogger, ignored k8s.IgnoreWrapper, serviceManager ServiceManager, configRefreshChan chan string) *Handler {
-	h := &Handler{
+func NewHandler(log logrus.FieldLogger, serviceManager ServiceManager, configRefreshChan chan struct{}) *Handler {
+	return &Handler{
 		log:               log,
-		ignored:           ignored,
 		configRefreshChan: configRefreshChan,
 		serviceManager:    serviceManager,
 	}
-
-	if err := h.Init(); err != nil {
-		log.Errorln("Could not initialize MeshControllerHandler")
-	}
-
-	return h
-}
-
-// Init handles any handler initialization.
-func (h *Handler) Init() error {
-	h.log.Debugln("MeshControllerHandler.Init")
-
-	return nil
 }
 
-// OnAdd executed when an object is added.
+// OnAdd is called when an object is added.
 func (h *Handler) OnAdd(obj interface{}) {
-	// assert the type to an object to pull out relevant data
-	switch obj := obj.(type) {
-	case *corev1.Service:
-		if h.ignored.IsIgnored(obj.ObjectMeta) {
-			return
-		}
-
+	// If the created object is a service we have to create a corresponding shadow service.
+	if obj, isService := obj.(*corev1.Service); isService {
 		if err := h.serviceManager.Create(obj); err != nil {
 			h.log.Errorf("Could not create mesh service: %v", err)
 		}
-	case *corev1.Endpoints:
-		return
-	case *corev1.Pod:
-		if !isMeshPod(obj) {
-			return
-		}
 	}
 
 	// Trigger a configuration rebuild.
-	h.configRefreshChan <- k8s.ConfigMessageChanRebuild
+	h.configRefreshChan <- configMessageChanRebuild
 }
 
-// OnUpdate executed when an object is updated.
+// OnUpdate is called when an object is updated and ensures that the proper handler is called depending on the filter matches.
```

Comment:
```suggestion
// OnUpdate is called when an object is updated.
```

There is no filtering anymore